opencensus/stats/internal/measure_data.cc

```diff
@@ -0,0 +1,89 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "opencensus/stats/internal/measure_data.h"
+
+#include <algorithm>
+#include <cmath>
+#include <cstdint>
+#include <iostream>
+#include <vector>
+
+#include "absl/base/macros.h"
+#include "absl/types/span.h"
+#include "opencensus/stats/bucket_boundaries.h"
+#include "opencensus/stats/distribution.h"
+
+namespace opencensus {
+namespace stats {
+
+MeasureData::MeasureData(absl::Span<const BucketBoundaries> boundaries)
+    : boundaries_(boundaries) {
+  histograms_.reserve(boundaries_.size());
+  for (const auto& b : boundaries_) {
+    histograms_.emplace_back(b.num_buckets());
+  }
+}
+
+void MeasureData::Add(double value) {
+  // Update using the method of provisional means.
+  ++count_;
+  ABSL_ASSERT(count_ > 0 && "Histogram count overflow.");
+  const double new_mean = mean_ + (value - mean_) / count_;
+  sum_of_squared_deviation_ =
```

Comment:
+=

Reply:
Updating sum_of_squared_deviation requires both the new and old mean. I guess it is a bit clearer to save the old mean locally (formerly in the next line count_ is the new count and mean_ the old mean), but I do not think it saves anything.